core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1499,4 +1499,18 @@ class ParserTest extends munit.ScalaCheckSuite {
     }
   }
 
+  property("p.filter(_ => true) == p") {
```

Comment:
Since branch is a new fundamental operation I think we need to add it to the generators. That will test to make sure none of the past properties are broken by it's presence.